src/main/frontend/src/app/appSlice.js

```diff
@@ -34,29 +27,39 @@ export const appSlice = createSlice({
     changeSelectedCourse: (state, action) => {
       state.selectedCourse = action.payload;
     },
+    setErrorFetchingCourses: (state, action) => {
+      state.errorFetchingCourses = action.payload;
+    },
   },
 });
 
 // Defines the actions that can be dispatched using dispatch and defines what happens with the state.
-export const { changeSearchInput, changeSelectedCourse, setLoading, setCourseArray, updateMetadata } = appSlice.actions;
+export const { changeSearchInput, changeSelectedCourse, setErrorFetchingCourses, setLoading, setCourseArray, updateMetadata } = appSlice.actions;
 // Connects variables to the state, when you want the state values in a component use this.
 export const selectSearchInputText = (state) => state.searchInputText;
 export const selectSelectedCourse = (state) => state.selectedCourse;
 export const selectCourseArray = (state) => state.filteredCourseArray;
+export const selectMetadata = (state) => state.metadata;
 export const selectIss = (state) => state.iss;
 export const selectContext = (state) => state.context;
 export const selectClientId = (state) => state.clientId;
 export const selectDeploymentId = (state) => state.deploymentId;
 export const selectIdToken = (state) => state.id_token;
 export const selectLoading = (state) => state.loading;
+export const selectErrorFetchingCourses = (state) => state.errorFetchingCourses;
 
 // This function fetches the courses from the backend, it should be invoked when loading the application.
-export const fetchCourses = (state) => (dispatch) => {
+export const fetchCourses = (page) => (dispatch, getState) => {
+  // Get the id_token from the state, thunks already have access to getState as the second argument.
```

Comment:
Nitpick: I think "thunks" is a typo

Reply:
I think it's correct Mary, it's a thunk function.

https://redux.js.org/usage/writing-logic-thunks